### PR TITLE
[angular] Sonar: add IDs for aria-describedby

### PR DESCRIPTION
**Pull request by qmonmert**
End of #13109 

cc @kaidohallik 

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-garbagecollector/metrics-garbagecollector.component.html.ejs [hunk 1]
```diff
@@ -65,7 +65,7 @@
         </div>
     </div>
 
-    <div class="col-md-4">
+    <div id="garbageCollectorMetrics" class="col-md-4">
```
Comment (kaidohallik) [inline generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-garbagecollector/metrics-garbagecollector.component.html.ejs:68]: This is unrelated to table from where `aria-describedby` refers. Maybe move title `metrics.jvm.gc.title` from `metrics.component.html` to this component and refer to that header?

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs [hunk 2]
```diff
@@ -50,8 +50,8 @@
         </span>
     </div>
 
-    <div class="pad" *ngFor="let thread of getThreads()">
-        <h6>
+    <div id="modalThreadsMetrics" class="pad" *ngFor="let thread of getThreads()">
+        <h6 id="modalThreadsMetrics{{ thread.threadId }}">
             <span class="badge" [ngClass]="getBadgeClass(thread.threadState)">{{ thread.threadState }}</span>
 
             &nbsp;{{ thread.threadName }} (ID {{ thread.threadId }})
```
Comment (kaidohallik) [inline generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs:57]: Here `id="modalThreadsMetrics"` is not best reference for `aria-describedby`.  
`id="modalThreadsMetrics{{ thread.threadId }}"` is unused.  
Maybe put `{{ thread.threadName }}` into `span` and give it id which includes `{{ thread.threadId }}` and in table use the same id in `aria-describedby` attribute?

**Review by kaidohallik — CHANGES_REQUESTED** [review]

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs (hunk 2 above)
Comment (kaidohallik) [reply generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs:57]: One possible solution is to follow Sonar instructions:

```
In order to be accessible to visually impaired users, it is important that tables provides a description of its content before the data is accessed.

The simplest way to do it, and also the one recommended by WCAG2 is to add a <caption> element inside the <table>.
```

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-garbagecollector/metrics-garbagecollector.component.html.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-garbagecollector/metrics-garbagecollector.component.html.ejs:70]: done

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs (hunk 2 above)
Comment (qmonmert) [reply generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs]: done

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics.component.html.ejs [hunk 3]
```diff
@@ -47,8 +47,6 @@
     </div>
 
     <div *ngIf="metrics && metricsKeyExists('garbageCollector')">
-        <h3 <%= jhiPrefix %>Translate="metrics.jvm.gc.title">Garbage collector statistics</h3>
-
         <<%= jhiPrefixDashed %>-metrics-garbagecollector
             [updating]="updatingMetrics"
             [garbageCollectorMetrics]="metrics.garbageCollector">
```
Comment (kaidohallik) [inline generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics.component.html.ejs:52]: Now it looks cleaner without outer `div` like other components are used below, can you do also this polishing:
```
    <<%= jhiPrefixDashed %>-metrics-garbagecollector
        *ngIf="metrics && metricsKeyExists('garbageCollector')"
        [updating]="updatingMetrics"
        [garbageCollectorMetrics]="metrics.garbageCollector">
    </<%= jhiPrefixDashed %>-metrics-garbagecollector>
```

File: generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs [hunk 4]
```diff
@@ -51,7 +51,7 @@
     </div>
 
     <div class="pad" *ngFor="let thread of getThreads()">
-        <h6>
+        <h6 id="modalThreadsMetrics{{ thread.threadId }}">
```
Comment (kaidohallik) [inline generators/client/templates/angular/src/main/webapp/app/admin/metrics/metrics-modal-threads/metrics-modal-threads.component.html.ejs:54]: Can you remove this added `id` - this is not used.

**Review by kaidohallik — CHANGES_REQUESTED** [review]

**Review by kaidohallik — APPROVED** [review]
LGTM, thanks @qmonmert
